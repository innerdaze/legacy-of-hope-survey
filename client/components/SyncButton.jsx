import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Button from 'grommet/components/Button'
import RefreshIcon from 'grommet/components/icons/base/Refresh'

export default class SyncButton extends Component {
  constructor(props) {
    super(props)

    this.sync = this.sync.bind(this)
  }

  sync() {
    this.props.sync()
  }

  render() {
    return (
      <Button
        plain
        icon={<RefreshIcon/>}
        onClick={this.sync}
        />
    )
  }
}

SyncButton.propTypes = {
  sync: PropTypes.func.isRequired
}

SyncButton.defaultProps = {
  sync: Function.prototype
}
