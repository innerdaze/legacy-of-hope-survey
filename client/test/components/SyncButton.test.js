import React from 'react'
import SyncButton from '../../components/SyncButton'

let wrapper, sync

describe('Testing SyncButton', () => {
  beforeEach(() => {
    sync = jest.fn()

    wrapper = shallow(<SyncButton sync={sync} />)
  })

  describe('SyncButton layout', () => {
    test('Expect to have a Button', () => {
      expect(wrapper.find('Button')).toHaveLength(1)

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('Test on prop', () => {
    test('Expect sync prop function to called if Button click', () => {
      wrapper.find('Button').simulate('click')
      expect(sync).toBeCalled()

      expect(wrapper).toMatchSnapshot()
    })
  })
})
