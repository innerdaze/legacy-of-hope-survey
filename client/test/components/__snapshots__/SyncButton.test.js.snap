// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SyncButton SyncButton layout Expect to have a Button 1`] = `
<Button
  icon={
    <Refresh
      responsive={true}
    />
  }
  method="push"
  onClick={[Function]}
  plain={true}
  type="button"
/>
`;

exports[`Testing SyncButton Test on prop Expect sync prop function to called if Button click 1`] = `
<Button
  icon={
    <Refresh
      responsive={true}
    />
  }
  method="push"
  onClick={[Function]}
  plain={true}
  type="button"
/>
`;
